datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaTopicPartitionTracker.java

```diff
@@ -68,12 +74,48 @@ public void onPartitionsRevoked(@NotNull Collection<TopicPartition> topicPartiti
         }
       }
     });
+
+    // Remove consumer offsets for partitions that have been revoked
+    topicPartitions.forEach(topicPartition -> {
+      Map<Integer, Long> partitions = _consumerOffsets.get(topicPartition.topic());
+      if (partitions != null) {
+        partitions.remove(topicPartition.partition());
+        if (partitions.isEmpty()) {
+          _consumerOffsets.remove(topicPartition.topic());
+        }
+      }
+    });
+  }
+
+  /**
+   * Updates consumer offsets for partitions that have been polled
+   * @param consumerRecords consumer records that have been the result of the poll
+   */
+  public void onPartitionsPolled(@NotNull ConsumerRecords<?, ?> consumerRecords) {
+    Collection<TopicPartition> topicPartitions = consumerRecords.partitions();
+
+    topicPartitions.forEach(topicPartition -> {
+      List<? extends ConsumerRecord<?, ?>> partitionRecords = consumerRecords.records(topicPartition);
+      ConsumerRecord<?, ?> lastRecord = partitionRecords.get(partitionRecords.size() - 1);
+
+      Map<Integer, Long> partitionOffsetMap = _consumerOffsets.putIfAbsent(topicPartition.topic(),
+          new ConcurrentHashMap<>());
+      partitionOffsetMap.put(topicPartition.partition(), lastRecord.offset());
```

Comment:
Here I'm making an assumption that record collection is ordered by offset, and also that it can't be empty. Need to ensure it's true.

Reply:
How are we testing these changes?

Reply:
I added unit tests.